libbeat/outputs/mode/lb/async_worker.go

```diff
@@ -111,25 +111,10 @@ func (w *asyncWorker) sendLoop() (done bool) {
 }
 
 func (w *asyncWorker) onMessage(msg eventsMessage) error {
-	var err error
 	if msg.datum.Event != nil {
-		err = w.client.AsyncPublishEvent(w.handleResult(msg), msg.datum)
-	} else {
-		err = w.client.AsyncPublishEvents(w.handleResults(msg), msg.data)
+		return w.client.AsyncPublishEvent(w.handleResult(msg), msg.datum)
 	}
-
-	if err != nil {
-		if msg.attemptsLeft > 0 {
-			msg.attemptsLeft--
-		}
-
-		// asynchronously retry to insert message (if attempts left), so worker can not
-		// deadlock on retries channel if client puts multiple failed outstanding
-		// events into the pipeline
-		w.onFail(msg, err)
-	}
-
-	return err
+	return w.client.AsyncPublishEvents(w.handleResults(msg), msg.data)
```

Comment:
Change in semantics is here. We always require the `w.client` instance to use the callback build via `handleResult` to report success/failure within a batch. This allows the output client to decide on sync or async error reporting. 
The async client as provided by `go-lumber` does require full async reporting, but the Logstash output did some sync reporting as well, leading to duplicates. PR changes to have `go-lumber` only trigger the async reporting, indirectly via `msgRef`.